cluster.go

```diff
@@ -1003,6 +1004,24 @@ func (c *Cluster) StateSync(ctx context.Context) error {
 			continue
 		}
 
+		var t time.Time
+		if !currentPin.PinOptions.ExpireAt.Equal(t) && currentPin.PinOptions.ExpireAt.Before(time.Now()) {
+			logger.Infof("StateSync: unpinning %s: pin expired", pCid)
+			var pinObj api.Pin
+			err := c.rpcClient.CallContext(
+				ctx,
+				getClosestTrustedPeer(ctx, pCid.String(), currentPin.Allocations, c.consensus),
+				"Cluster",
+				"Unpin",
+				currentPin,
+				&pinObj,
+			)
+			if err != nil {
+				logger.Warning(err)
+			}
```

Comment:
Don't randomly hand off to other peers. If you have 1000 peers, one of them will receive 1000 requests to unpin a pin that has expired. What you want to figure out is "am I the peer most likely to be in charge of unpinning this item?" and then proceed if that is so.